src/conn/conn_compact.c

```diff
@@ -80,6 +80,47 @@ __background_compact_exclude_list_clear(WT_SESSION_IMPL *session, bool closing)
         __wt_free(session, conn->background_compact.exclude_list_hash);
 }
 
+/*
+ * __background_compact_exclude_list_process --
+ *     Process the exclude list present in a configuration.
+ */
+static int
+__background_compact_exclude_list_process(WT_SESSION_IMPL *session, const char *config)
+{
+    WT_CONFIG exclude_config;
+    WT_CONFIG_ITEM cval, k, v;
+    WT_DECL_RET;
+    const char *cfg[3] = {NULL, NULL, NULL};
+
+    cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_compact);
+    cfg[1] = config;
+    cfg[2] = NULL;
+
+    __background_compact_exclude_list_clear(session, false);
+    WT_RET_NOTFOUND_OK(__wt_config_gets(session, cfg, "exclude", &cval));
+    if (cval.len != 0) {
+        /*
+         * Check that the configuration string only has table schema formats in the target list
+         * and construct the target hash table.
+         */
+        __wt_config_subinit(session, &exclude_config, &cval);
+        while ((ret = __wt_config_next(&exclude_config, &k, &v)) == 0) {
+            if (!WT_PREFIX_MATCH(k.str, "table:"))
+                WT_RET_MSG(session, EINVAL,
```

Comment:
As I read it this appears to leave the compact exclude list in a unpredictable state if you encounter a non`table:` entry in the config: it might be empty, it may contain the first N valid entries.

Reply:
I agree, this will be done in WT-12066.